packages/core/src/io/node-io.ts

```diff
@@ -34,31 +34,49 @@ import { PlatformIO } from './platform-io';
 export class NodeIO extends PlatformIO {
 	private _fs;
 	private _path;
+	private _fetch;
+	private _httpRegex = /https?:\/\//;
+	public useFetch = false;
 
 	/** Constructs a new NodeIO service. Instances are reusable. */
-	constructor() {
+	// eslint-disable-next-line @typescript-eslint/no-explicit-any
+	constructor(fetch?: any) {
 		super();
 		// Excluded from browser builds with 'package.browser' field.
 		this._fs = require('fs').promises;
 		this._path = require('path');
+		this._fetch = fetch;
+    this.useFetch = !!fetch;
 	}
 
 	protected async readURI(uri: string, type: 'view'): Promise<Uint8Array>;
 	protected async readURI(uri: string, type: 'text'): Promise<string>;
 	protected async readURI(uri: string, type: 'view' | 'text'): Promise<Uint8Array | string> {
 		switch (type) {
-			case 'view':
+			case 'view': {
+        if(this.useFetch && this._httpRegex.exec(uri)) {
+					const response = await this._fetch(uri);
+					return new Uint8Array(await response.arrayBuffer());
+				} 
 				return this._fs.readFile(uri);
-			case 'text':
+      }
+			case 'text': {
+        if(this.useFetch && this._httpRegex.exec(uri)) {
+					const response = await this._fetch(uri);
+					return await response.text();
+				} 
 				return this._fs.readFile(uri, 'utf8');
+      }
 		}
 	}
 
 	protected resolve(directory: string, path: string): string {
-		return this._path.resolve(directory, path);
+    if(this.useFetch) return `${directory}/${path}`;
+		return this._path.join(directory, path);
```

Comment:
I can't quite tell from the CI output why the tests are failing, but I think it might be this method. One important difference between `path.resolve` and `path.join` is that `resolve` will ignore the directory if the path is already absolute. Perhaps the logic here should be:

1. If `directory` or `path` matches httpRegex, use the [same resolver as WebIO](https://github.com/donmccurdy/glTF-Transform/blob/290e028542e1c23fe10c4f940a84b6181a5e3b72/packages/core/src/io/web-io.ts#L68-L87)
2. If neither matches httpRegex, use `_path.resolve` as before.

In retrospect, probably `basePath` or `rootPath` would have been better names than `directory` here.